core/plugins/impl/normalize_plugin.cpp

```diff
@@ -0,0 +1,292 @@
+std::string NormalizePlugin::serializeToString() const {
+  torch::serialize::OutputArchive output_archive;
+  std::vector<int64_t> axesVec(axes_.begin(), axes_.end());
+  output_archive.write("order", torch::IValue((int64_t)order_));
+  output_archive.write("axes", torch::IValue(axesVec));
+  output_archive.write("keep_dims", torch::IValue((int64_t)keep_dims_));
+  std::ostringstream data_str;
+  output_archive.save_to(data_str);
+
+  return data_str.str();
+}
+
+size_t NormalizePlugin::getSerializationSize() const {
+  return serializeToString().size();
+}
+
+bool NormalizePlugin::supportsFormatCombination(
+    int pos,
+    const nvinfer1::PluginTensorDesc* inOut,
+    int nbInputs,
+    int nbOutputs) {
+  TRTORCH_ASSERT(0 <= pos && pos <= 1, "There should be exactly 2 connections to the plugin - 1 input, 1 output");
+  TRTORCH_ASSERT(nbInputs == 1, "Expected a single tensor as input to normalize plugin");
+  TRTORCH_ASSERT(nbOutputs == 1, "Expected a single tensor as output to normalize plugin");
+
+  const nvinfer1::PluginTensorDesc& in = inOut[0];
+
+  if (pos == 0) {
+    return (in.type == nvinfer1::DataType::kFLOAT) && (in.format == nvinfer1::TensorFormat::kLINEAR);
+  }
+
+  // pos == 1, accessing information about output tensor
+  const nvinfer1::PluginTensorDesc& out = inOut[1];
+
+  return (in.type == out.type) && (in.format == out.format);
+}
+
+void NormalizePlugin::configurePlugin(
+    const nvinfer1::DynamicPluginTensorDesc* in,
+    int nbInputs,
+    const nvinfer1::DynamicPluginTensorDesc* out,
+    int nbOutputs) {
+  dtype_ = nvinfer1::DataType::kFLOAT;
+}
+
+size_t NormalizePlugin::getWorkspaceSize(
+    const nvinfer1::PluginTensorDesc* inputs,
+    int nbInputs,
+    const nvinfer1::PluginTensorDesc* outputs,
+    int nbOutputs) const {
+  return 0;
+}
+
+int NormalizePlugin::enqueue(
+    const nvinfer1::PluginTensorDesc* inputDesc,
+    const nvinfer1::PluginTensorDesc* outputDesc,
+    const void* const* inputs,
+    void* const* outputs,
+    void* workspace,
+    cudaStream_t stream) {
+  // TRT <= 7.0
+#if NV_TENSORRT_MAJOR < 7 || (NV_TENSORRT_MAJOR == 7 && NV_TENSORRT_MINOR < 1)
+  at::Tensor input = at::from_blob((void*)inputs[0], util::toVec(inputDesc->dims), [](void*) {}, tensor_options_);
+  at::Tensor output = at::from_blob(outputs[0], util::volume(outputDesc->dims), [](void*) {}, tensor_options_);
+
+  at::cuda::CUDAStream torch_stream = at::cuda::getStreamFromPool();
+  at::cuda::CUDAStreamGuard torch_guard(torch_stream);
+
+  cudaEvent_t event;
+  cudaEventCreate(&event);
+  cudaEventRecord(event, stream);
+
+  cudaStreamWaitEvent(torch_stream.stream(), event, 0);
+  at::Tensor result = at::norm(input, order_, axes_, keep_dims_);
+  output.copy_(result);
+  cudaEvent_t torch_event;
+  cudaEventCreate(&torch_event);
+  cudaEventRecord(torch_event, torch_stream.stream());
+
+  cudaStreamWaitEvent(stream, torch_event, 0);
+
+  cudaEventDestroy(event);
+  cudaEventDestroy(torch_event);
+  return 0;
+#else

```

Comment:
Are we still stuck with CPU execution for ATen?

Reply:
I did not try with GPU execution yet.